packages/devtools_app/lib/src/memory/memory_heap_tree_view.dart

```diff
@@ -505,13 +502,14 @@ class HeapTreeViewState extends State<HeapTree>
                               // We're collapsing close the fields table.
                               controller.selectedLeaf = null;
                             }
-                            setState(() {});
+                            setState(() {
+                              controller.buildTreeFromAllData();
```

Comment:
is this setState call needed? we shouldn't have to wrap calls to the controller with a setState. Is there another piece of data from the controller we need the widget to listen on?

Reply:
I've fixed all the setState calls.